musica-fortran/src/micm/MICM_mod.F90

```diff
@@ -0,0 +1,69 @@
+module libmicm
+    use iso_c_binding
+
+    private
+    public :: micm
+
+    include "MICM_C_def.F90"
+
+    type micm
+        private
+        type(c_ptr) :: ptr
+    contains
+        procedure :: delete => delete_micm_polymorph
+        procedure :: create_solver => micm_create_solver
+        procedure :: solve => micm_solve
+    end type
+
+    interface micm
+        procedure create_micm
+    end interface
+
+contains
+    function create_micm(config_path)
+        implicit none
+        type(micm) :: create_micm
+        character(len=*), intent(in) :: config_path
+        character(len=1, kind=C_CHAR) :: c_config_path(len_trim(config_path) + 1)
+        integer :: N, i
+
+        ! Converting Fortran string to C string
+        N = len_trim(config_path)
+        do i = 1, N
+            c_config_path(i) = config_path(i:i)
+        end do
+        c_config_path(N + 1) = C_NULL_CHAR
+
+        create_micm%ptr = create_micm_c(c_config_path)
+    end function
+
+    subroutine delete_micm(this)
+        implicit none
+        type(micm) :: this
+        call delete_micm_c(this%ptr)
+    end subroutine
+    
+    subroutine delete_micm_polymorph(this)
+        implicit none
+        class(micm) :: this
+        call delete_micm_c(this%ptr)
+    end subroutine
```

Comment:
Is this needed?

Reply:
My understanding is that yes, we need this. I also had the same question and without it, it doesn't work. It needs to declare micm as `class` instead of `type` to be able to have access to its members.  I found [this answer](https://stackoverflow.com/questions/53421609/polymorphism-in-fortran) useful to understand.

Reply:
I updated it to `class` and removed `type` version.